pkg/envinfo/service.go

```diff
@@ -0,0 +1,57 @@
+package envinfo
```

Comment:
why is this in `envinfo`? THis package is handling `env.yaml` file, the function in this file are working on top of `devfile.yaml`

Reply:
That's because I was going through the way things are done for adding URL & storage information to devfile. It's being done from `envinfo` package right now.

Reply:
But since the code in this PR is not dependent on `env.yaml` in any way, I get why putting it here would be a bad idea. In that case, putting it under `pkg/service` seems like the way to me. WDYT?

Just looked at this again and currently "odo service create" calls `AddServiceToDevfile` from `pkg/service/service.go` whereas `odo service delete` calls `DeleteServiceFromDevfile` from the newly created `pkg/odo/cli/service/backends.go`. And it doesn't delete a Service Catalog based service from devfile.

Reply:
I feel we should just rename the package name to something which indicates `env.yaml + devfile.yaml`as that might help with abstraction.

Reply:
it would help if we separate abstraction, devfile and env.yaml are two completely different things they should not live in one package

Reply:
But devfile already has the library. I don't feel we need a separate package to add `odo` concepts like service, url, storage etc to it. Since both are some sort of local data storage, combining them makes more sense to me. The outside packages like CLI don't need to know where the data comes from locally. That's how the URL create, delete etc for local are designed currently.

Reply:
The function that are here in this file, don't have anything specific to odo, expect its name.
`AddServiceToDevfile` Could be named `AddK8sComponentToDevfile`, use the same parameters and it would make sense as well (maybe even more sense)

Reply:
> The function that are here in this file, don't have anything specific to odo, expect its name.
> `AddServiceToDevfile` Could be named `AddK8sComponentToDevfile`, use the same parameters and it would make sense as well (maybe even more sense)

Since it has nothing specific to odo and doesn't fit into the local config provider interface, I would suggest making it a utils function or a private function. I don't think we need a seperate package for it.

Reply:
> `AddServiceToDevfile` Could be named `AddK8sComponentToDevfile`

I like this suggestion. Makes more sense since from devfile point of view we're actually adding a k8s component.

> I would suggest making it a utils function or a private function. I don't think we need a seperate package for it.

If we rename it to `AddK8sComponentToDevfile`, it sure doesn't make sense to move this to `pkg/service` either. It's more generic than being a part of `pkg/service`. Which utils package would this make more sense to live in - `pkg/util` or `pkg/odo/util`? And why (asking this for better understanding on my part)?

Reply:
It can also be part of some utils function in the `pkg/service` package as crds are mostly handled by it.

Reply:
PTAL https://github.com/openshift/odo/pull/4465/commits/f4123527811c23c04fdf97799a6dca797c510dfc. I have kept it in `pkg/service` but not a part of utils yet since I made the changes before you made a comment.

Reply:
Looks OK to me